src/decoder_allocators.hpp

```diff
@@ -41,16 +41,16 @@ namespace zmq
     class c_single_allocator
     {
     public:
-        c_single_allocator(size_t bufsize_):
+        explicit c_single_allocator(size_t bufsize_):
                 bufsize(bufsize_),
-                buf((unsigned char*) malloc (bufsize))
+                buf(static_cast<unsigned char*>( malloc (bufsize) ))
```

Comment:
I have not seen a verdict on spacing type modifiers, `T* x` vs `T *x`. If we don't get a style rule on this, I recommend going with "majority per file".